src/gc/gc.cpp

```diff
@@ -26685,7 +26683,7 @@ void gc_heap::background_grow_c_mark_list()
         assert (new_c_mark_list);
         memcpy (new_c_mark_list, c_mark_list, c_mark_list_length*sizeof(uint8_t*));
         c_mark_list_length = c_mark_list_length*2;
-        delete c_mark_list;
+        delete [] c_mark_list;
```

Comment:
@Maoni0 Sure.  Do you want me to revert this change and do it as part of the valgrind exercise #10800?  Or just create an issue now since we know it is 'broken'?